opal/corelib/array.rb

```diff
@@ -63,7 +76,15 @@ def initialize(size = nil, obj = nil, &block)
   end
 
   def self.new(*args, &block)
-    [].initialize(*args, &block)
+    obj = allocate
+    breaker = obj.initialize(*args, &block)
+    %x{
+      if (obj.$$has_breaker !== undefined) {
```

Comment:
Yes, that's my assumption here, `Array#initialize` **usually** returns self. And I'm not sure if it's a good idea to rely on it :smile: